dpypelines/pipeline/db/dataset_statuses_collection.py

```
@@ -0,0 +1,177 @@
+        :param new_status: The status type associated with the update event
+        :param upload_event: Whether the update event is related to uploading to an external service (DatasetAPI or Upload Service)
+
+        :return: models.DatasetStatus
+        """
+        # Get current status
+        current_status_model = self.get_status(status_oid)
+
+        # Append new event to current_status.events
+        if current_status_model:
+            current_status_model.events.append(event)
+
+            # Generate `update_values` dict to pass to `update_one_document` method
+            update_values = {
+                "updated_at": event.timestamp.isoformat(),
+                "status": new_status.name,
+                "events": [
+                    event.dict_for_mongodb() for event in current_status_model.events
+                ],
+                "error_message": event.error_message,
+                "retry_count": event.retry_count,
+                "last_retry_timestamp": event.last_retry_timestamp,
+            }
+
+            # Handle upload events
+            if (
+                upload_event
+                and event.event_data
+                and event.event_data.upload_location
+                == models.UploadLocation.DATASET_API
+            ):
+                update_values["uploaded_to_dataset_api"] = True
+            elif (
+                upload_event
+                and event.event_data
+                and event.event_data.upload_location
+                == models.UploadLocation.UPLOAD_SERVICE
+            ):
+                update_values["uploaded_to_upload_service"] = True
+
+            # Update status document with new values
+            self.__collection.update_one_document({"_id": status_oid}, update_values)
+
+        # Get updated status document and convert to DatasetStatus model
+        updated_status_model = self.get_status(status_oid)
+        if updated_status_model:
```

Maybe raise an error here if not?